cabal-install-solver/src/Distribution/Solver/Types/PkgConfigDb.hs

```diff
@@ -63,10 +68,15 @@ readPkgConfigDb verbosity progdb = handle ioErrorHandler $ do
     case mpkgConfig of
       Nothing             -> noPkgConfig "Cannot find pkg-config program"
       Just (pkgConfig, _) -> do
-        pkgList <- lines <$> getProgramOutput verbosity pkgConfig ["--list-all"]
+        -- To prevent malformed Unicode in the descriptions from crashing cabal,
+        -- read without interpreting any encoding first. (#9608)
+        pkgList <- LBS.split (fromIntegral (ord '\n')) <$>
+                     getProgramInvocationLBS verbosity (programInvocation pkgConfig ["--list-all"])
         -- The output of @pkg-config --list-all@ also includes a description
         -- for each package, which we do not need.
-        let pkgNames = map (takeWhile (not . isSpace)) pkgList
+        let pkgNamesLBS = map (LBS.takeWhile (not . isSpace . chr . fromIntegral)) pkgList
+        -- Now decode as UTF8 and convert to String, dropping any that fail decoding.
+        let pkgNames = rights $ map (fmap T.unpack . T.decodeUtf8' . LBS.toStrict) pkgNamesLBS
```

Comment:
Dropping things without giving any output saying that you're doing that seems likely to lead to very hard-to-debug issues later. Can we log (at high verbosity) saying that we're ignoring an entry because it has invalid unicode?

Reply:
Thanks! You're right. I'm now logging something at `Verbose` level, which seems to be `-v2` (but not 100% sure). The test also checks this.